### PR TITLE
link/term: .dvc/ (dir) & "internal"

### PR DESCRIPTION
**Pull request by jorgeorpinel**
> Continues #2098 ~~and #2117~~

- ~~ Rename section about `foreach` in dvc.yaml (remove term "multi-stage").~~ Extracted to #2124
- [x] Autolink `.dvc/` to https://dvc.org/doc/user-guide/project-structure/internal-files.
- [x] Review term "internal".
- ~~Review mentions of `.dvcignore` and auto-link it.~~ Extracted to #2125

### CODE REVIEW

File: content/docs/user-guide/project-structure/pipelines-files.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ ${param.list[0]} # List elements via index in [] (square brackets)
 > To use the expression literally in `dvc.yaml` (so DVC does not replace it for
 > a value), escape it with a backslash, e.g. `\${...`.
 
-## Generating multiple stages
+## Foreach stages
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/pipelines-files.md:227]: Does everyone agree on this term? Cc @dberenbaum

File: content/linked-terms.js [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+  },
   {
     matches: '.dvc',
     url: '/doc/user-guide/project-structure/dvc-files'
   },
   {
-    matches: 'dvc.yaml',
-    url: '/doc/user-guide/project-structure/pipelines-files'
+    matches: '.dvcignore',
+    url: '/doc/user-guide/project-structure/dvcignore-files'
```
Comment (jorgeorpinel) [inline content/linked-terms.js:16]: Hi @rogermparent 👋 

For some reason this new autolink is not being picked up in https://dvc-landing-2-0-pipelin-xkejno.herokuapp.com/doc/user-guide/project-structure.

It works everywhere in https://dvc-landing-2-0-pipelin-xkejno.herokuapp.com/doc/command-reference/add and https://dvc-landing-2-0-pipelin-xkejno.herokuapp.com/doc/command-reference/check-ignore.

I notices that if the bullet doesn't start with `` `.dvcignore` `` then it works. But that's not a problem for the previous bullets which start with `` `dvc.yaml` `` and `` `.dvc` `` 🤷

File: content/docs/user-guide/project-structure/pipelines-files.md (hunk 1 above)
Comment (shcheklein) [reply content/docs/user-guide/project-structure/pipelines-files.md:227]: `foearch` should be code-inlined I think ... There was an option like "Something (`foreach`)" - I would personally prefer that one

**Comment by shcheklein** [comment]
PR again addressed to much stuff at once

**Comment by jorgeorpinel** [comment]
Yeah sorry, #2098 was a significant change that required all these link updates, and reviewing those, better options started to emerge (like the auto-links). I didn't plan all the check-boxes in advance, just added them as they came.

**Comment by shcheklein** [comment]
@jorgeorpinel 👍 but let's try to this as 4 PRs please :) it's very hard to review this.

File: content/linked-terms.js (hunk 2 above)
Comment (rogermparent) [reply content/linked-terms.js:16]: Looks like it's being picked up to me. Maybe it was a cache issue?

File: content/docs/user-guide/project-structure/pipelines-files.md (hunk 1 above)
Comment (dberenbaum) [reply content/docs/user-guide/project-structure/pipelines-files.md:227]: I can't find any suggestion like "Something (`foreach`)" but I'm fine with "`foreach` stages"

File: content/linked-terms.js (hunk 2 above)
Comment (jorgeorpinel) [reply content/linked-terms.js:16]: Yep. Thanks @rogermparent

Comment (jorgeorpinel) [reply content/linked-terms.js:16]: Well, it still happens on my local even after `rm -rf .cache/` but I see it's OK in the review app 🤷

File: content/docs/user-guide/project-structure/pipelines-files.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/pipelines-files.md]: I think the "Something" was "Loop" but I argued we should avoid that direct reference.

Went for quoted **`foreach` stages** — a minor issue is how it looks in the ToC nav:

image

**Comment by jorgeorpinel** [comment]
> let's try to this as 4 PRs please

Done. Order: #2124, this one, #2125

**Comment by jorgeorpinel** [comment]
> let's try to this as 4 PRs please

Done @shcheklein. Order: #2124, this one, #2125

**Review by shcheklein — APPROVED** [review]

**Pull request merged**